h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -119,9 +124,9 @@ private GridSearch(Key<Grid> gkey, HyperSpaceWalker<MP, ?> hyperSpaceWalker) {
     long gridWork=0;
     if (gridSize > 0) {//if total grid space is known, walk it all and count up models to be built (not subject to time-based or converge-based early stopping)
       int count=0;
-      while (it.hasNext(model) && (it.max_models() > 0 && count++ < it.max_models())) { //only walk the first max_models models, if specified
+      while (it.hasNext(Optional.ofNullable(model)) && (it.max_models() > 0 && count++ < it.max_models())) { //only walk the first max_models models, if specified
         try {
-          Model.Parameters parms = it.nextModelParameters(model);
+          Model.Parameters parms = it.nextModelParameters(Optional.of(model));
```

Comment:
so model was nullable 2 lines above, and it isn't anymore?

Reply:
this comment needs to be resolved

Reply:
I completely reverted all the changes to Hyperspace walkers, leaving the original behavior.